src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx

```diff
@@ -227,6 +294,221 @@ export function GeneralEntityRoot<T extends BaseEntity>({
     }
   };
 
+  const onBeforeFileUpload = (file: RcFile) => {
+    // TODO: Complete quality check
+    // 1. Check for file size. What should the limit be? Format dependent?
+    // 2. Check if the selected file format is valid
+    // 3. Check if the file is not broken/corrupted
+    // 4. Other checks
```

Comment:
Generally this would be possible I think, but to reduce complexity and duplication of code, we decided to make use of the GeoServer REST endpoints directly. If not already done in GS, I would favour to add the necessary validity checks to it instead of implementing a SHOGun wrapper.

Reply:
also ok for me :+1: